core/src/main/java/hudson/util/RobustReflectionConverter.java

```diff
@@ -451,6 +451,13 @@ private boolean fieldDefinedInClass(Object result, String attrName) {
 
     protected Object unmarshalField(final UnmarshallingContext context, final Object result, Class type, Field field) {
         Converter converter = mapper.getLocalConverter(field.getDeclaringClass(), field.getName());
+        if (converter == null) {
```

Comment:
See https://github.com/jenkinsci/jenkins/pull/9727#discussion_r1761868912 for discussion about cases where `converter` is not null. In short, I don't think we care. 

I am not sure though whether this method covers all of the cases we care about. It seems like we may need similar changes related to "implicit collections". I will look into that a bit.

Reply:
Specifically, prior to this line, I think we should look up the field and then add code comparable to what we added to `RobustCollectionConverter` to validate the value type: https://github.com/jenkinsci/jenkins/blob/1836bd6ef4cece1e261dcce22f57dab3cc898c7f/core/src/main/java/hudson/util/RobustReflectionConverter.java#L477

Needs to be investigated with a test case.

Reply:
I looked into the cases related to implicit collections today. My understanding is that you need to explicitly call `addImplicitArray`, `addImplicitCollection`, or `addImplicitMap` for the relevant code to matter, which is not very common, see [here](https://github.com/search?type=code&q=+owner%3Ajenkinsci+addImplicit+-path%3Asrc%2Ftest+-repo%3Ajenkinsci%2Fjs-libs+-repo%3Ajenkinsci%2Fgroovy+-repo%3Ajenkinsci%2Fxstream+-repo%3Ajenkinsci%2Fxstream-fork) (there are also a few usages in CloudBees plugins). The overloads where you specify the item type already avoid JENKINS-63343 because of `getFieldNameForItemTypeAndName` [here](https://github.com/jenkinsci/jenkins/blob/1836bd6ef4cece1e261dcce22f57dab3cc898c7f/core/src/main/java/hudson/util/RobustReflectionConverter.java#L458) which uses `context.getRequiredType()` leading to invalid elements being silently ignored. The overloads for arrays and maps seem to not actually work at all due to [this check](https://github.com/jenkinsci/jenkins/blob/1836bgd6ef4cece1e261dcce22f57dab3cc898c7f/core/src/main/java/hudson/util/RobustReflectionConverter.java#L466-L469) (which leaves me somewhat confused about https://github.com/jenkinsci/jenkins/pull/2621, but perhaps my testing did not cover all cases).

So, I think we do not really care much about implicit collections, but [09ca355](https://github.com/jenkinsci/jenkins/pull/9727/commits/09ca355d0eaa8212129dab7aeb0acd1d16e9fcf5) should handle them too.